js/style/style.js

```diff
@@ -472,6 +467,10 @@ Style.prototype = util.inherit(Evented, {
         this.fire(e.type, util.extend({source: e.target}, e));
     },
 
+    _forwardLayerEvent: function(e) {
+        this.fire(e.type, util.extend({layer: e.target.id}, e));
```

Comment:
> Are you avoiding `{layer: e.target}` because `StyleLayer` objects are otherwise not part of the public API?

Exactly. 

> `{layer: {id: e.target.id}}` would be more future-proof.

:+1: